File: solutions/NMF/Benchmark/ISolution.cs
using NMF.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NMFSolution.Benchmark
{
    public interface ISolution
    {
        void Initialize();
        Func<Model> ComputeChanges(string modelPath, string model, int iteration, string targetPath);
        Model Initial(string modelPath, string model, string targetPath);
        void Load(string modelPath, string model);
    }
}

File: solutions/emfSolution/src/main/java/ttc2025/ISolution.java
package ttc2025;

import DOT.Graph;

import java.util.function.Supplier;

/**
 * Denotes a solution to the benchmark
 */
public interface ISolution {
    /**
     * initializes the solution
     */
    void initialize() throws Exception;

    /**
     * Loads the model
     *
     * @param modelPath the path of the model
     * @param model the name of the model
     */
    void load(String modelPath, String model) throws Exception;

    /**
     * Performs the initial transformation
     *
     * @param modelPath the path of the model
     * @param model the name of the model
     * @param targetPath the full path to the dot file, in case the solution directly creates dot files
     * @return a graph, in case an in-memory graph is returned, or null
     */
    Graph initial(String modelPath, String model, String targetPath) throws Exception;

    /**
     * Calculates the change propagations
     *
     * @param modelPath the path of the model
     * @param model the name of the model
     * @param iteration the iteration
     * @param targetPath the full path to the dot file, in case the solution directly creates dot files
     * @return a lambda expression that will perform the change propagation
     */
    Supplier<Graph> computeChanges(String modelPath, String model, int iteration, String targetPath) throws Exception;
}
